core/src/main/scala/org/apache/spark/scheduler/LiveListenerBus.scala

```diff
@@ -32,18 +33,24 @@ import org.apache.spark.util.Utils
  * has started will events be actually propagated to all attached listeners. This listener bus
  * is stopped when `stop()` is called, and it will drop further events after stopping.
  */
-private[spark] class LiveListenerBus extends SparkListenerBus {
+private[spark] class LiveListenerBus(val sparkContext: SparkContext) extends SparkListenerBus {
```

Comment:
I guess we also use this to tear down `sparkContext` when the listener dies. I still have a weak preference for the old code, however, since I think the lifecycle was clearer.